mythril/laser/ethereum/strategy/basic.py

```diff
@@ -1,54 +1,80 @@
 """
 This module implements basic symbolic execution search strategies
 """
+from abc import ABC, abstractmethod
+from random import randrange
 
+try:
+    from random import choices
+except ImportError:
+
+    from itertools import accumulate
+    from random import random
+    from bisect import bisect
+
+    def choices(population, weights=None):
+        if weights is None:
+            return [population[int(random() * len(population))]]
+        cum_weights = accumulate(weights)
+        return [population[bisect(cum_weights, random()*cum_weights[-1], 0, len(population)-1)]]
+
+
+class BasicStrategy(ABC):
```

Comment:
I'd name this SearchStrategy.
Also can you put this in __init__ ? I think that is a better place for this class

Reply:
Let’s add the ‘basic’ term as a prefix as not all search strategies can be emulated with this class.how about calling it something like ‘BasicSearchStrategy’.